enhancements/20231113-witness-node-support.md

```diff
@@ -0,0 +1,71 @@
+# Support witness node
+
+## Summary
+
+The witness node is a lightweight node that only runs the control plane components. The witness node is not schedulable and does not run any workload. The witness node forms a quorum with the other two nodes.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/3266
+
+## Motivation
+
+### Goals
+
+Create a three-node cluster with one witness node.
+
+## Proposal
+
+### User Stories
+
+For the edge use case, the witness node support benefits both cost and high availability. The classical scenario would be two nodes and one witness node.
+
+### User Experience In Detail
+
+1. Set the node label `node-role.harvesterhci.io/etcd=true` to the node.
+2. The promoter will add the node to the cluster as `etcd` role.
+
+### API changes
+
+None
+
+## Design
+
+### Implementation Overview
+
+#### Promote controller
+
+Harvester depends on the job `promote` to promote a node to related role after it joins (the default role is worker). We can only promote the node to `etcd` role with the label `node-role.harvesterhci.io/etcd=true`.
+
+After this feature, the harvester node will have the following roles:
+1. master node roles:
+   - node-role.kubernetes.io/control-plane: "true"
+   - node-role.kubernetes.io/etcd: "true"
+   - node-role.kubernetes.io/master: "true"
+
+2. etcd node roles:
+   - node-role.kubernetes.io/etcd: "true"
+
+3. worker node roles:
+
+#### Installation
+
+1. Interactive mode:
+   - We need to add an extra panel for setting the witness node (only for join cluster mode).
+
+2. Non-interactive mode:
+   - We can simply the label `node-role.harvesterhci.io/etcd=true` to the node by yaml.
```

Comment:
How about adding a new mode in https://docs.harvesterhci.io/v1.2/install/harvester-configuration#installmode? It's more consistent.

Reply:
How about?
```
install:
  - mode
  - role
```

Reply:
How about `mode: witness`? just add another role there.

Reply:
The `mode` is more like an operation because we use `create`/`join`.
So, I prefer to add `role` for the specific node role.